backend/tests/app/datasets/test_service.py

```diff
@@ -137,6 +137,18 @@ def test_search_dataset_matching_name_and_description(self):
         assert results[0].description == ds.description
         assert results[0].rank == pytest.approx(0.5833333)
 
+    def test_search_dataset_partial_prefix_matches(self):
```

Comment:
Can we also add a test to see if `clin data` matches this `Clinical dataset` datasets? To check that the matching works with multiple words?